File: python/백준/재귀/2448 별 찍기 - 11.py
import sys
input = sys.stdin.readline

def star(n, r, c):
    if n == 1:
        return '*'
    
    # 현재 위치가 어느 블록에 속하는지
    block = n // 3
    if (r // block) == 1 and (c // block) == 1:
        return ' '  # 가운데 블록은 공백
    
    return star(block, r % block, c % block)

n = int(input())

result = []
for r in range(n):
    row = []
    for c in range(n):
        row.append(star(n, r, c))
    result.append(''.join(row))

print('\n'.join(result))

File: cpp/백준/재귀/2448 별 찍기 - 11.cpp
#include <iostream>
#include <string>
using namespace std;

char star(int n, int r, int c) {
    if (n == 1) return '*';

    int block = n / 3;
    if ((r / block) == 1 && (c / block) == 1) {
        return ' ';
    }
    return star(block, r % block, c % block);
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    for (int r = 0; r < n; ++r) {
        string row;
        row.reserve(n);
        for (int c = 0; c < n; ++c) {
            row += star(n, r, c);
        }
        cout << row << "\n";
    }
    return 0;
}
